src/commerce-handler.js

```diff
@@ -9,66 +10,143 @@ CommerceHandler.prototype.logCommerceEvent = function(event) {
             }
-        } else {
-            optimizelyEvent.eventName = expandedEvent.EventName;
-            if (
-                event.CustomFlags &&
-                event.CustomFlags['Optimizely.EventName']
-            ) {
-                optimizelyEvent.eventName =
-                    event.CustomFlags['Optimizely.EventName'];
+
+            // Events that are added to the OptimizelyUI will be available on optimizelyWebXEvents.events
+            // Ignore events not included in the Optimizely UI
+            if (optimizelyWebXEvents.events[optimizelyWebXEvent.eventName]) {
+                var eventCopy = {};
+                for (var key in optimizelyWebXEvent) {
+                    eventCopy[key] = optimizelyWebXEvent[key];
+                }
+                window['optimizely'].push(eventCopy);
             }
         }
 
-        // Events that are added to the OptimizelyUI will be available on optimizelyEvents.events
-        // Ignore events not included in the Optimizely UI
-        if (optimizelyEvents.events[optimizelyEvent.eventName]) {
-            var eventCopy = {};
-            for (var key in optimizelyEvent) {
-                eventCopy[key] = optimizelyEvent[key];
+        // if optimizely full stack is being used
+        if (window.optimizelyClientInstance) {
+            if (optimizelyFullStackEvents.events[expandedEvent.EventName]) {
+                var eventKey = event.EventName,
+                userId,
+                userAttributes = this.common.userAttributes,
+                eventTags = {};
+
+                eventTags = expandedEvent.EventAttributes || {};
+
+                if (window.mParticle && window.mParticle.Identity) {
+                    var userIdentities = window.mParticle.Identity.getCurrentUser().getUserIdentities()['userIdentites'];
+        
+                    switch(this.forwarderSettings.userIdField) {
+                        case 'customerId':
+                            userId = userIdentities["customerId"];
+                            break;
+                        case 'email':
+                            userId = userIdentities["email"];
+                            break;
+                        case 'mpid':
+                            userId = userIdentities["mpid"];
+                            break;
+                        case 'other':
+                            userId = userIdentities["other"];
+                            break;
+                        case 'other2':
+                            userId = userIdentities["other2"];
+                            break;
+                        case 'other3':
+                            userId = userIdentities["other3"];
+                            break;
+                        case 'other4':
+                            userId = userIdentities["other4"];
+                            break;
+                        default:
+                            userId = null;
+                    }
+                }
```

Comment:
so, `forwarderSettings.userIdField` is not accessible here. add `userIdField` to `common` in initialization, then that will be accessible here from `self.common.userIdeField`

then, add a boolean for `userIdIsSet`, (or something else that makes sense), and set to true at the end of the switch statement, and above 

```suggestion
if (!self/this.common.userId) {
                    switch(this.forwarderSettings.userIdField) {
                        case 'customerId':
                            userId = userIdentities["customerId"];
                            break;
                        case 'email':
                            userId = userIdentities["email"];
                            break;
                        case 'mpid':
                            userId = userIdentities["mpid"];
                            break;
                        case 'other':
                            userId = userIdentities["other"];
                            break;
                        case 'other2':
                            userId = userIdentities["other2"];
                            break;
                        case 'other3':
                            userId = userIdentities["other3"];
                            break;
                        case 'other4':
                            userId = userIdentities["other4"];
                            break;
                        default:
                            userId = null;
                    }
                    self/this.common.userId = userId;
                }
}
```

same logic applies to eventhandler.js